src/legacy/core_plugins/kibana/public/dashboard/plugin.ts

```diff
@@ -90,12 +86,11 @@ export class DashboardPlugin implements Plugin {
       stateParams: [
         {
           kbnUrlKey: '_g',
-          stateUpdate$: querySyncStateContainer.state$,
+          stateUpdate$: data.query.global$,
```

Comment:
This is the main improvement. The issue described in https://github.com/elastic/kibana/issues/56503